### PR TITLE
New Package: cmatrix

### PR DESCRIPTION
**Pull request by techie-guy**
Adds cmatrix to msys2

### CODE REVIEW

File: cmatrix/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,33 @@
+_realname='cmatrix'
+pkgname="${_realname}-git"
+pkgver=2.0
+pkgrel=2
+pkgdesc="A curses-based scrolling 'Matrix'-like screen"
+arch=('any')
+url="http://www.asty.org/cmatrix/"
+license=('GPL3')
+depends=('ncurses' 'make' 'autotools')
+makedepends=('git')
```
Comment (lazka) [inline cmatrix/PKGBUILD:10]:
```suggestion
depends=('ncurses')
makedepends=('git' 'make' 'autotools' 'gcc')
```

Comment (techie-guy) [reply cmatrix/PKGBUILD]: Ah, I forgot gcc. Thanks!

**Comment by techie-guy** [comment]
Builds successfully now:)

**Comment by techie-guy** [comment]
Uses Source from the cmatrix Release Page

File: cmatrix/define_TIOCSTI.patch [hunk 2]
```diff
@@ -0,0 +1,15 @@
+diff --git a/cmatrix.c b/cmatrix.c
+index 90284cb..7bca3f3 100644
+--- a/cmatrix.c
++++ b/cmatrix.c
+@@ -57,6 +57,10 @@
+ #include <termio.h>
+ #endif
+ 
++#ifdef __CYGWIN__
++#define TIOCSTI 0x5412
```
Comment (Biswa96) [inline cmatrix/define_TIOCSTI.patch:10]: Should it be defined here? Does cygwin/msys2 support TIOCSTI ioctl?

Comment (techie-guy) [reply cmatrix/define_TIOCSTI.patch:10]: It is defined in the master branch of cmatrix
https://github.com/abishekvashok/cmatrix/blob/master/cmatrix.c#L74

It was not defined in the release version 2.0, So I added a patch to define it

Removing it and adding `#include <sys/ioctl.h>`, gives this error
```sh
cmatrix.c:488:41: error: ‘TIOCSTI’ undeclared
```

Comment (techie-guy) [reply cmatrix/define_TIOCSTI.patch:10]: What should I do?

Comment (Biswa96) [reply cmatrix/define_TIOCSTI.patch:10]: Just asking. I am little fearful about this change.

Comment (techie-guy) [reply cmatrix/define_TIOCSTI.patch:10]: okay

**Comment by Biswa96** [comment]
Some suggestions.
1. Please remove _realname, pkgbase, conflicts and provides.
2. The depends should have ncurses only. The rest would be in makedepends.

**Comment by techie-guy** [comment]
Done.
If there's anything else, Please let me know.

**Comment by Biswa96** [comment]
Thank you for the contribution 🎉

**Pull request merged**